src/core/api/chart_repository.go

```diff
@@ -347,18 +426,36 @@ func (cra *ChartRepositoryAPI) DeleteChart() {
 	// Get other parameters from the request
 	chartName := cra.GetStringFromPath(nameParam)
 
+	key := fmt.Sprintf("chart:%s:%s", cra.namespace, chartName)
+	lock, err := requireLock(key)
```

Comment:
Should lock to the version level?

Reply:
The `chart:namespace:chartName` will be locked when upload, delete the chart version now, so when delete chart, no lock needed for the chart version.

And the resource will be free when chart deleted success.